### PR TITLE
fix: Adding OTEL_EXPORTER_OTLP_ENDPOINT

### PR DESCRIPTION
**Pull request by zacharyblasczyk**
This pr 
* Fixes the pg and pg-pool tracing 
* Adds implements the logging with `winston-transport`
* Updates the docker-compose files to work with otel if desired. 
* Fix the broken docker-compose.yaml

## Release Notes

- **New Features**
	- Introduced a new OpenTelemetry Collector service configuration for enhanced telemetry data collection.
	- Added support for OpenTelemetry in the logging system for improved observability.
  
- **Updates**
	- Updated various service configurations in Docker files to streamline dependency management and environment variables.
	- Enhanced the instrumentation setup by switching to a simpler span processing model.

- **Bug Fixes**
	- Removed duplicate entries and unnecessary dependencies to optimize performance.

- **Documentation**
	- Updated package dependencies to reflect the latest versions and ensure compatibility.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -26,52 +26,79 @@ services:
...
       - postgres
 
   webservice:
-    platform: linux/amd64
-    image: ctrlplane/webservice:c656deb
+    # platform: linux/amd64
+    # image: ctrlplane/webservice:75685e0
+    build:
+      context: .
+      dockerfile: ./apps/webservice/Dockerfile
```
Comment (zacharyblasczyk) [inline docker-compose.yaml:42]: Just `docker-compose build` if you wanna test locally.

File: docker-compose.yaml [hunk 2]
```diff
@@ -26,52 +26,79 @@ services:
...
+
+      AUTH_GOOGLE_CLIENT_ID: 27586067691-1us5sc17f0pq0vs4c0ml1eaqphmf9sl9.apps.googleusercontent.com
+      AUTH_GOOGLE_CLIENT_SECRET: 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
 
       NEXTAUTH_URL: ${BASE_URL}
-      DATABASE_URL: postgres://ctrlplane:ctrlplane@postgres:5432/ctrlplane
+      OTEL_EXPORTER_OTLP_ENDPOINT: "http://otel-collector:4318"
+      REDIS_URL: redis://redis:6379
+      POSTGRES_URL: postgresql://ctrlplane:ctrlplane@postgres:5432/ctrlplane
+      VARIABLES_AES_256_KEY: d2506d733ee210078461c08ee8e1605be75ed73b0941d4b513ab7b159c6fbcd9
```
[the remark this answers is not in the document]
Comment (zacharyblasczyk) [reply docker-compose.yaml:59]: This is just a sample key for demonstration.

File: docker-compose.yaml [hunk 3]
```diff
@@ -46,32 +50,53 @@ services:
       # AUTH_OIDC_CLIENT_SECRET: "<oidc-client-secret>"
 
       NEXTAUTH_URL: ${BASE_URL}
-      DATABASE_URL: postgres://ctrlplane:ctrlplane@postgres:5432/ctrlplane
+      OTEL_EXPORTER_OTLP_ENDPOINT: "http://otel-collector:4318"
+      REDIS_URL: redis://redis:6379
+      POSTGRES_URL: postgres://ctrlplane:ctrlplane@postgres:5432/ctrlplane
+      VARIABLES_AES_256_KEY: d2506d733ee210078461c08ee8e1605be75ed73b0941d4b513ab7b159c6fbcd9
+      # NODE_OPTIONS: "--require /app/apps/webservice/.next/serverinstrumentation.js"
```
[the remark this answers is not in the document]
Comment (zacharyblasczyk) [reply docker-compose.yaml]: Not a secret.

File: apps/webservice/Dockerfile [hunk 4]
```diff
@@ -20,6 +20,7 @@ COPY turbo.json turbo.json
 RUN pnpm add -g turbo
 
 COPY package.json package.json
+COPY .npmrc .npmrc
```
Comment (zacharyblasczyk) [inline apps/webservice/Dockerfile:23]: @jsbroks this is the fix.

File: apps/webservice/Dockerfile [hunk 5]
```diff
@@ -59,6 +60,7 @@ EXPOSE 3000
 ENV PORT=3000
 ENV AUTH_TRUST_HOST=true
 ENV NODE_ENV=production
+ENV NODE_PATH=/app/apps/webservice/node_modules
```
Comment (jsbroks) [inline apps/webservice/Dockerfile:63]: Why do we need this @zacharyblasczyk

Comment (zacharyblasczyk) [reply apps/webservice/Dockerfile:63]: I don't think we need it. Cleaning it up and testing.

**Review by jsbroks — APPROVED** [review]

**Pull request merged**